PeriodogramAnalysis/PowerSpectrumAnalysis.py

```diff
@@ -59,139 +56,147 @@ def __call__(self, signal: SignalType):
 
         Returns:
         --------
-        tuple
-            PSD dictionary, excluded channels, number of chunks, band display range, dB conversion flag, spectrogram dictionary, frequency limit.
+        psd_dict:
+            PSD dictionary
+        spec_dict
+            spectrum dictionary
         """
-        for idx, signal_piece in enumerate(signal):
-            if idx >= 1:
-                break
+        psd_dict = defaultdict(dict)
+        spec_dict = defaultdict(dict)
+        for chunk_index, signal_piece in enumerate(signal):
+            self.chunk_idx = chunk_index
             self.num_channels = signal_piece.number_of_channels
```

Comment:
It is bad practice: Don't define class variable within the member function. It become hard to trace where these variables are coming from.

Reply:
Get it. I have removed them.